### ISSUE TITLE
Markdown processor seems confused when using links

### ISSUE THREAD

**Issue by bitnapper** [internetarchive/openlibrary#3714]
When mixing links and other **bold** or _italic_ text in the _Work Description_, the interpreter seems to omit some of the formatters as you can see in the following links.

### Evidence / Screenshot

- [Wenn die Sterne verlöschen](https://openlibrary.org/books/OL28812924M/)
- [Die Verschwender vom Mars](https://openlibrary.org/books/OL28818736M/)

### Details

- **Logged in (Y/N)?** not relevant
- **Browser type/version?**  not relevant
- **Operating system?**  not relevant
- **Environment (prod/dev/local)?** prod

**Comment by guyjeangilles** [comment]
@bitnapper please post a screenshot of the exact markdown used and the result of it's rendering.

**Comment by bitnapper** [comment]
![4837D872-A1C1-4469-9780-6B894BF735CD](https://user-images.githubusercontent.com/2772669/90474988-df533300-e126-11ea-951e-8c35e45767f7.jpeg)
Here‘s the Markdown


![61E22CA0-01F7-4D73-8F26-2876871D2037](https://user-images.githubusercontent.com/2772669/90474985-de220600-e126-11ea-9586-adfa22749323.jpeg)
Here‘s the result.

**Comment by bitnapper** [comment]
![2727DFB3-2AFC-4890-B1F4-83A26DB07268](https://user-images.githubusercontent.com/2772669/90475186-47097e00-e127-11ea-8909-7fc3017f1dd3.jpeg)

**Comment by bitnapper** [comment]
![4D12C25D-43D6-4503-9DBD-806BF1A076D6](https://user-images.githubusercontent.com/2772669/90475343-964fae80-e127-11ea-85ef-f19468bba505.jpeg)
Here in widescreen with the complete Markdown.

**Comment by miphilomath** [comment]
I would like to work on this issue if someone can guide me a little. I looked up the code base to understand which files are handling the markdown. 

Found out, **openlibrary/core/olmarkdown.py** was importing following modules

```
import re
from infogami.utils.markdown import markdown
from openlibrary.core import helpers as h
... 
...
class OLMarkdown(markdown.Markdown):
  ...
  def convert(self):
          html = markdown.Markdown.convert(self)
          return h.sanitize(html)
```

But the module markdown was missing which were imported, that could be the root of the issue.

**Comment by seabelis** [comment]
There's also a problem with formatting in the edition description and TOC. All formatting is ignored.

**Comment by mekarpeles** [comment]
@miphilomath yes please! Your help would be appreciated and @jimchamp and I can help as you have questions

**Comment by cdrini** [comment]
Based on the comment I think this might've been meant to be assigned to you @jimchamp :)

**Comment by mekarpeles** [comment]
It seems the problem is that previews work irrespective of where `*` are however the renderer only works when the `*` is outside (I think) the `[]`'s. In any case, it either doesn't work when inside or when outside.

The resolutions are either to make this work at the render level or to disable the preview for the not-working case.

@miphilomath does either approach seems easier / more reasonable to you and do you have any further questions which @jimchamp can help with?
